elyra/pipeline/pipeline.py

```diff
@@ -440,3 +406,73 @@ def __eq__(self, other: "Pipeline") -> bool:
                 and self.runtime_config == other.runtime_config
                 and self.operations == other.operations
             )
+
+
+class KeyValueList(list):
+    """
+    A list class that exposes functionality specific to lists whose entries are
+    key-value pairs separated by a pre-defined character.
+    """
+
+    _key_value_separator: str = "="
+
+    def to_dict(self, logger: Optional[Logger] = None) -> Dict[str, str]:
+        """
+        Properties consisting of key/value pairs are stored in a list of separated
+        strings, while most processing steps require a dictionary - so we must convert.
+        If no key/value pairs are specified, an empty dictionary is returned, otherwise
+        pairs are converted to dictionary entries and returned.
+        """
+        kv_dict = {}
+        for kv in self:
+            if not kv:
+                continue
+
+            if self._key_value_separator not in kv:
+                raise ValueError(
+                    f"Property {kv} does not contain the expected "
+                    f"separator character: '{self._key_value_separator}'."
+                )
+
+            key, value = kv.split(self._key_value_separator, 1)
+            if not key.strip():
+                KeyValueList.log_message(f"Skipping inclusion of property '{kv}': no key found", logger, logging.WARN)
+                continue
+            if not value:
+                KeyValueList.log_message(
+                    f"Skipping inclusion of property '{key}': no value specified", logger, logging.INFO
```

Comment:
I'm not sure `INFO` is the right severity for this one - thinking more `DEBUG`.  The reason being is that one of these log messages will be produced for every environment variable name that is programmatically parsed from the generic component since none of those results includes a value (and assuming the user didn't follow up on that by manually adding a value).  When `DEBUG` is not enabled, I'm not sure it's useful information since that is essentially default behavior.

Reply:
Yeah, I agree. I'm actually surprised I never noticed this being logged at INFO before now (when this functionality resided in `Operation`